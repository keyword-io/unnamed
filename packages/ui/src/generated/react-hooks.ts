import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
export type Maybe<T> = T | null;
export type InputMaybe<T> = Maybe<T>;
export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
const defaultOptions = {} as const;
/** All built-in and custom scalars, mapped to their actual values */
export type Scalars = {
  ID: string;
  String: string;
  Boolean: boolean;
  Int: number;
  Float: number;
  /** A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar. */
  DateTime: any;
};

export type Mutation = {
  __typename?: 'Mutation';
  createDraft?: Maybe<Post>;
  deletePost?: Maybe<Post>;
  incrementPostViewCount?: Maybe<Post>;
  signupUser: User;
  togglePublishPost?: Maybe<Post>;
};


export type MutationCreateDraftArgs = {
  authorEmail: Scalars['String'];
  data: PostCreateInput;
};


export type MutationDeletePostArgs = {
  id: Scalars['Int'];
};


export type MutationIncrementPostViewCountArgs = {
  id: Scalars['Int'];
};


export type MutationSignupUserArgs = {
  data: UserCreateInput;
};


export type MutationTogglePublishPostArgs = {
  id: Scalars['Int'];
};

export type Post = {
  __typename?: 'Post';
  author?: Maybe<User>;
  content?: Maybe<Scalars['String']>;
  createdAt: Scalars['DateTime'];
  id: Scalars['Int'];
  published: Scalars['Boolean'];
  title: Scalars['String'];
  updatedAt: Scalars['DateTime'];
  viewCount: Scalars['Int'];
};

export type PostCreateInput = {
  content?: InputMaybe<Scalars['String']>;
  title: Scalars['String'];
};

export type PostOrderByUpdatedAtInput = {
  updatedAt: SortOrder;
};

export type Query = {
  __typename?: 'Query';
  allUsers: Array<User>;
  draftsByUser?: Maybe<Array<Maybe<Post>>>;
  feed: Array<Post>;
  postById?: Maybe<Post>;
};


export type QueryDraftsByUserArgs = {
  userUniqueInput: UserUniqueInput;
};


export type QueryFeedArgs = {
  orderBy?: InputMaybe<PostOrderByUpdatedAtInput>;
  searchString?: InputMaybe<Scalars['String']>;
  skip?: InputMaybe<Scalars['Int']>;
  take?: InputMaybe<Scalars['Int']>;
};


export type QueryPostByIdArgs = {
  id?: InputMaybe<Scalars['Int']>;
};

export enum SortOrder {
  Asc = 'asc',
  Desc = 'desc'
}

export type User = {
  __typename?: 'User';
  email: Scalars['String'];
  id: Scalars['Int'];
  name?: Maybe<Scalars['String']>;
  posts: Array<Post>;
};

export type UserCreateInput = {
  email: Scalars['String'];
  name?: InputMaybe<Scalars['String']>;
  posts?: InputMaybe<Array<PostCreateInput>>;
};

export type UserUniqueInput = {
  email?: InputMaybe<Scalars['String']>;
  id?: InputMaybe<Scalars['Int']>;
};

export type AllUsersQueryVariables = Exact<{ [key: string]: never; }>;


export type AllUsersQuery = { __typename?: 'Query', allUsers: Array<{ __typename?: 'User', id: number, name?: string | null }> };


export const AllUsersDocument = gql`
    query allUsers {
  allUsers {
    id
    name
  }
}
    `;

/**
 * __useAllUsersQuery__
 *
 * To run a query within a React component, call `useAllUsersQuery` and pass it any options that fit your needs.
 * When your component renders, `useAllUsersQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useAllUsersQuery({
 *   variables: {
 *   },
 * });
 */
export function useAllUsersQuery(baseOptions?: Apollo.QueryHookOptions<AllUsersQuery, AllUsersQueryVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<AllUsersQuery, AllUsersQueryVariables>(AllUsersDocument, options);
      }
export function useAllUsersLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<AllUsersQuery, AllUsersQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<AllUsersQuery, AllUsersQueryVariables>(AllUsersDocument, options);
        }
export type AllUsersQueryHookResult = ReturnType<typeof useAllUsersQuery>;
export type AllUsersLazyQueryHookResult = ReturnType<typeof useAllUsersLazyQuery>;
export type AllUsersQueryResult = Apollo.QueryResult<AllUsersQuery, AllUsersQueryVariables>;