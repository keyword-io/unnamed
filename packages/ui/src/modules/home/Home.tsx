import React, { useEffect, useState } from 'react';
import List from '../../components/List';
import { useAllUsersQuery } from '../../generated/react-hooks';

const Home: React.FC = props => {
  const [list, setList] = useState<Array<{ id: string; label: string }>>([]);

  useEffect(() => {
    setList([
      {
        id: '1',
        label: 'item-1',
      },
      {
        id: '2',
        label: 'item-2',
      },
    ]);
  }, []);

  const { loading, error } = useAllUsersQuery({
    onCompleted(data) {
      setList(
        data.allUsers?.map(user => ({
          id: user.id.toString(),
          label: user.name || '',
        }))
      );
    },
  });

  function render() {
    if (loading) {
      return <h1>loading...</h1>;
    }

    if (error) {
      return <h1>error: {error.message}</h1>;
    }

    return <List list={list} />;
  }

  return <main>{render()}</main>;
};

export default Home;
