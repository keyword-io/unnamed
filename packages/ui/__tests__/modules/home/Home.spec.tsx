import React from 'react';
import {render, screen} from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Home from '../../../src/modules/home/Home';
import { MockedProvider } from "@apollo/client/testing";
import { AllUsersDocument } from '../../../src/generated/react-hooks'

test('loads and displays greeting', async () => {
  const mocks = [
    {
      request: {
        query: AllUsersDocument,
      },
      result: {
        data: {
          allUsers: [
            {
              id: '1',
              name: 'item-1'
            },
            {
              id: '2',
              name: 'item-2'
            }
          ]
        }
      }
    }
  ];

  const { container } = render(
    <MockedProvider mocks={mocks}>
      <Home />
    </MockedProvider>
  )
  expect(container).toMatchSnapshot();

  expect(await screen.findByText('loading...')).toBeInTheDocument()

  const item1 = await screen.findByText('item-1');
  expect(item1).toBeInTheDocument();

  const listItems = document.querySelectorAll('.list-item')
  expect(listItems.length).toBe(2);

  await userEvent.click(item1);
  expect(item1.className.indexOf('__active') > -1).toBeTruthy();
})
