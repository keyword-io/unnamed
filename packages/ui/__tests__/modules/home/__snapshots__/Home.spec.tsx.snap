// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads and displays greeting 1`] = `
<div>
  <main>
    <h1>
      loading...
    </h1>
  </main>
</div>
`;
